.github/workflows/build-push.yml

```diff
@@ -0,0 +1,335 @@
+            else
+              BUILD_BRANCH="$ORIGIN_BRANCH"
+            fi
+          fi
+          
+          echo "Build branch: $BUILD_BRANCH"
+          echo "BUILT_BRANCH_NAME=$BUILD_BRANCH" >> $GITHUB_ENV
+
+      - name: Validate branch strategy
+        run: |
+          ORIGIN_BRANCH="${{ github.ref_name }}"
+          
+          # Check if we would commit to the same branch as the origin
+          if [ "$ORIGIN_BRANCH" = "${{ env.BUILT_BRANCH_NAME }}" ]; then
+            echo "❌ ERROR: Build artifacts would be committed to the same branch as the source code!"
+            echo ""
+            echo "This workflow is designed to keep source code and build artifacts separate."
+            echo "You have two options:"
+            echo ""
+            echo "1. Use a 'dev/' prefixed branch (e.g., 'dev/main', 'dev/feature-branch')"
+            echo "   - Source: dev/main → Build artifacts: main"
+            echo "   - Source: dev/ABC-123 → Build artifacts: ABC-123"
+            echo ""
+            echo "2. Provide an explicit BUILT_BRANCH_NAME input to specify where build artifacts should go"
+            echo ""
+            echo "Current branch: $ORIGIN_BRANCH"
+            echo "Target build branch: ${{ env.BUILT_BRANCH_NAME }}"
+            echo ""
+            exit 1
+          fi
+          
+          echo "✅ Branch strategy validated:"
+          echo "  Source branch: $ORIGIN_BRANCH"
+          echo "  Build branch: ${{ env.BUILT_BRANCH_NAME }}"
+
+      - name: Checkout the build branch and merge in it
+        run: |
+          git show-ref -q refs/remotes/origin/${{ env.BUILT_BRANCH_NAME }} && git checkout ${{ env.BUILT_BRANCH_NAME }} || git checkout -b ${{ env.BUILT_BRANCH_NAME }}
+          git checkout ${{ github.ref_name }} -- .
+
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ inputs.PHP_VERSION }}
+          tools: php-scoper, sniccowp/php-scoper-wordpress-excludes, composer/installers, inpsyde/wp-translation-downloader, ${{ inputs.PHP_TOOLS }}
+          coverage: none
+
+      - name: Install Composer dependencies
+        uses: ramsey/composer-install@v3
+        env:
+          COMPOSER_AUTH: '${{ secrets.COMPOSER_AUTH_JSON }}'
+        with:
+          composer-options: ${{ inputs.COMPOSER_ARGS }}
+
+      - name: Set up node
+        uses: actions/setup-node@v4
+        env:
+          NODE_OPTIONS: ${{ inputs.NODE_OPTIONS }}
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_REGISTRY_TOKEN }}
+        with:
+          node-version: ${{ inputs.NODE_VERSION }}
+          registry-url: ${{ inputs.NPM_REGISTRY_DOMAIN }}
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Determine package type
+        run: |
+          if [ -f "style.css" ]; then
+            echo "Package type: WordPress Theme"
+            echo "IS_THEME=true" >> $GITHUB_ENV
+          else
+            echo "Package type: WordPress Plugin"
```

Comment:
What about Libraries?

Reply:
I suppose it's just a matter of wording. :) We have a few libraries that also ship JS/CSS and need the build workflow. 💪🏻

Reply:
I refactored things a little, so that we are only checking against a generic `PACKAGE_TYPE` from now on. This would make it easier to address types other than plugins/themes in the future.
https://github.com/inpsyde/reusable-workflows/pull/194/commits/d0deb15ae4ec6a34a6afd8cc847d958960ae1ee2